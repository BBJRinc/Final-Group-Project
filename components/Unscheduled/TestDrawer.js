import React, { Component } from 'react';
import { Drawer } from 'native-base';
import Unscheduled from './Unscheduled';

export default class TestDrawer extends Component{
    render(){
        
    }
}