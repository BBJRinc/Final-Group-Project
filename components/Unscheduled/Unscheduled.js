import React, { Component } from 'react';
import { ListView, StyleSheet, Modal } from 'react-native';
import { Container, Header, Content, Button, Icon, List, ListItem, Text, Left, Right, Body, Title, Footer, FooterTab } from 'native-base';
import FooterMenu from '../Footer/FooterMenu';
import axios from 'axios';


const datas = [
    {name: 'Simon Mignolet', age: 31},
    {name: 'Nathaniel Clyne', age: 23},
    {name: 'Dejan Lovren', age: 15},
    {name: 'Mama Sakho', age: 38},
    {name: 'Alberto Moreno', age: 31},
    {name: 'Emre Can', age: 56},
    {name: 'Joe Allen', age: 41},
    {name: 'Phil Coutinho', age: 32},
    
  ];

export default class Unscheduled extends Component{
    constructor(props) {
        super(props);
        this.ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });
        this.state = {
          basic: true,
          listViewData: [],
        };
      }

    componentDidMount(){
        axios.get('http://192.168.3.149:4040/api/unscheduled').then(resp => {
            this.setState({listViewData: resp.data});
        });
    }

    deleteTask(id){
        axios.delete(`http://192.168.3.149:4040/api/unscheduled/${id}`).then(resp => {
            console.log(resp.data);
            this.setState({listViewData: resp.data});
        });
    }

    render(){
        const ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });
        return(
            <Container>
                <Modal  
                    animationType="slide"
                    transparent={false}
                    visible={this.props.visible}
                    onRequestClose={() => this.props.showMenuItem('showTasks')}
                    >
                    <Header style={styles.header}>
                        <Left style={{width: 300}}>
                            <Title style={styles.black}>Unscheduled Tasks</Title>                                                
                        </Left>
                        <Right>
                            <Button transparent onPress={() => this.props.showMenuItem('showTasks')}>
                                <Icon name="close" style={styles.black}/>
                            </Button>
                        </Right>
                    </Header>
                    <Content>
                    <List
                        dataSource={this.ds.cloneWithRows(this.state.listViewData)}
                        renderRow={data =>
                        <ListItem onPress={() => this.props.onTaskPress(data, 'showTasks')}>
                            <Text> {data.taskname} </Text> 
                        </ListItem>}
                        renderLeftHiddenRow={data =>
                        <Button full onPress={() => this.props.onTaskPress(data, 'showTasks')}>
                            <Icon active name="information-circle" />
                        </Button>}
                        renderRightHiddenRow={(data, secId, rowId, rowMap) =>
                        <Button full danger onPress={() => this.deleteTask(data.taskid)}>
                            <Icon active name="trash" />
                        </Button>}
                        leftOpenValue={75}
                        rightOpenValue={-75}
                        />
                    </Content>
                </Modal>
            </Container>
        )
    }
}

const styles = StyleSheet.create({
    header: {
      backgroundColor: '#f2f2f2'
    },
    black: {
        color: 'black',
        fontSize: 18
    }
  });