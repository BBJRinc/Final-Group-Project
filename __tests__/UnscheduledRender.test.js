//Jordan's test

import React from 'react';
import Unscheduled from '../components/Unscheduled/Unscheduled';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<Unscheduled />).toJSON();
    expect(tree).toMatchSnapshot();
});