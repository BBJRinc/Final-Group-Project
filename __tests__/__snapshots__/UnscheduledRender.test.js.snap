// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    onShow={[Function]}
    transparent={false}
    visible={true}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F8F8F8",
              "borderBottomColor": "#a7a6ab",
              "borderBottomWidth": 0.5,
              "elevation": 3,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "left": 0,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 15,
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": "#000",
              "width": "100%",
            },
          ]
        }
        virtual={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "alignSelf": "center",
                "flex": 1,
              },
              Object {
                "flex": 1,
              },
            ]
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 17,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                  "fontSize": 18,
                },
              ]
            }
            virtual={undefined}
          >
            Unscheduled Tasks
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
          virtual={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": null,
                "borderLeftWidth": null,
                "borderRadius": 50,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "marginRight": -8,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#007aff",
                      "fontSize": 27,
                      "marginLeft": 2,
                      "marginRight": 2,
                      "marginTop": 2,
                      "paddingTop": 2,
                    },
                    Object {
                      "color": "#fff",
                      "fontSize": 18,
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
              virtual={undefined}
            >
              ÔêÜ
            </Text>
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={undefined}
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutoAutomaticScroll={true}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      viewIsInsideTabBar={false}
      virtual={undefined}
    >
      <View>
        <RCTScrollView
          closeOnRowBeginSwipe={false}
          closeOnRowPress={true}
          closeOnScroll={true}
          dataSource={
            ListViewDataSource {
              "items": 0,
            }
          }
          directionalDistanceChangeThreshold={2}
          disableLeftSwipe={false}
          disableRightSwipe={false}
          enableEmptySections={true}
          leftOpenValue={75}
          onScroll={[Function]}
          previewFirstRow={false}
          recalculateHiddenLayout={false}
          renderLeftHiddenRow={[Function]}
          renderRightHiddenRow={[Function]}
          renderRow={[Function]}
          renderScrollComponent={[Function]}
          rightOpenValue={-75}
          style={Object {}}
          swipeToOpenPercent={50}
          virtual={undefined}
        >
          <View />
        </RCTScrollView>
      </View>
    </RCTScrollView>
  </Modal>
</View>
`;
